src/rrddump.ml

```diff
@@ -40,11 +40,12 @@ let text_export rrd =
 
 let with_input_file filename f =
   let ic = open_in filename in
-finally (fun () -> f ic) (fun () -> close_in ic)
+  try f ic
+  with _ -> close_in ic
```

Comment:
This is not equivalent: if `f ic` succeeds then we leak `ic` instead of closing it.
You could probably do something like:
```
let result =
 try f ic
 with e ->
  begin try close_in ic with _ -> () end;
  raise e
 in
close_in ic;
result
```